addons/survey/static/src/js/survey_form.js

```diff
@@ -162,22 +176,119 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend({
             $("html, body").animate({ scrollTop: 0 }, "fast");
         }
         else if (result && result.fields && result.error === 'validation') {
-            var fieldKeys = _.keys(result.fields);
-            _.each(fieldKeys, function (key) {
-                self.$("#" + key + '>.o_survey_question_error').append($('<p>', {text: result.fields[key]})).toggleClass('d-none', false);
-                if (fieldKeys[fieldKeys.length - 1] === key) {
-                    self._scrollToError(self.$('.o_survey_question_error:visible:first').closest('.js_question-wrapper'));
-                }
+            self.$('.o_survey_form_content').fadeIn(400, function () {
+                self._showErrors(result.fields);
             });
             return false;
         } else {
             var $target = self.$('.o_survey_error');
-            $target.toggleClass('d-none', false);
+            $target.addClass("slide_in");
             self._scrollToError($target);
             return false;
         }
     },
 
+    // VALIDATION TOOLS
+    // -------------------------------------------------------------------------
+    _validateForm: function ($form, formData) {
+        var self = this;
+        var errors = {};
+        var validationErrorMsg = core._t("The answer you entered is not valid.");
```

Comment:
Isn't that something that is configurable on the backend side (`validation_error_msg`) ?
If so, it should be passed instead of redefined.